src/org/parosproxy/paros/model/SiteMap.java

```diff
@@ -745,5 +746,49 @@ private void clearParentFilter (SiteNode parent) {
 			clearParentFilter(parent.getParent());
 		}
 	}
+	
+	public void removeNodeFromParent(SiteNode node) {

```

Comment:
Probably safer to override the method removeNodeFromParent(MutableTreeNode) (we could validate that the node is a SiteNode).